src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,68 @@
+package backend;
+
+import util.Utility;
+
+import java.util.Optional;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoID {
+
+    private final String repoIDString;
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Return an Optional with a new instance of repoID if repoIDString is valid
+     * @param repoIDString
+     * @return Optional<RepoID>
+     */
+    public static Optional<RepoID> createRepoIDInstance(String repoIDString){
```

Comment:
It might be better to indicate the error with an exception instead of an Optional. The latter is generally for indicating that a value can be absent for reasons callers cannot control, and requiring that they handle that case. Here, it's the caller's responsibility to provide a valid string. We can reasonably expect that, so we don't have to make every caller go through the trouble of unwrapping the Optional.

Also, the name is a bit long. Maybe just `create`?